### PR TITLE
net: series of fix typo in multiple directories

### PR DESCRIPTION
**Pull request by ndrs-pst**
Typo corrections in various `subsys/net` directories:

- subsys/net/lib/lwm2m/*
- subsys/net/lib/l2/*
- subsys/net/lib/*

### CODE REVIEW

**Review by pdgendt — APPROVED** [review]

**Review by jukkar — APPROVED** [review]

**Review by maass-hamburg — APPROVED** [review]

**Review by awojasinski — APPROVED** [review]

**Review by ghost — APPROVED** [review]
For ieee802154: Thank you so much. Highly appreciated.

**Review by rlubos — APPROVED** [review]
Compliance doesn't seem to like British English, probably easiest just to apply the suggestion to resolve the issue. Looks good otherwise.

**Review by glarsennordic — APPROVED** [review]
LGTM, thanks!

File: subsys/net/l2/ieee802154/ieee802154_priv.h [hunk 1]
```diff
@@ -63,7 +63,7 @@ int ieee802154_wait_for_clear_channel(struct net_if *iface);
  *        This function has side effects and must be called before each individual transmission
  *        attempt.
  *
- *        This function checks for and supports both, software and hardware acknowlegement,
+ *        This function checks for and supports both, software and hardware acknowledgement,
```
Comment (nashif) [inline subsys/net/l2/ieee802154/ieee802154_priv.h:66]: @ndrs-pst  can you please fix that?

Comment (nashif) [reply subsys/net/l2/ieee802154/ieee802154_priv.h:66]: we can fix the spelling file later, both versions of the spelling are correct it seems.

**Pull request merged**